src/decoder.rs

```diff
@@ -49,77 +49,22 @@ pub enum Instr {
     Mret,
     Sret,
     Vfencevma,
-    /// Load Double-Word (register-based)
-    Ld {
+    /// Load (register-based)
+    Load {
         rd: Register,
         rs1: Register,
         imm: isize,
+        len: usize,
```

Comment:
We already have an enum with the possible cases here.

https://github.com/CharlyCst/miralis/blob/ad9f737ac97eab3242cffc5b4f38351398c4e534/src/device.rs#L9-L19

Currently it is in Device, maybe we could move it to `decoder.rs` if that makes more sense, but it would be better to re-use it throughout the whole code base rather than using different way of representing the width.

Reply:
Okay, won't it make more sense then to move the `Width` enum to `utils.rs` or other general-purpose module? I anticipate the enum to be used to some extent in `decoder`, `device`, `driver`, `metal` and `virt` modules.

Reply:
Then yes `util.rs` works, `arch/mod.rs` can work too, we already have some generic architecture definitions there, both sounds good to me :)